test/ft/di_config.cpp

```diff
@@ -68,6 +68,36 @@ test constructible_policy_must_be_bound = [] {
   injector.create<c>();
 };
 
+template <class>
+struct is_shared_ptr : std::false_type {};
+
+template <class T>
+struct is_shared_ptr<std::shared_ptr<T>> : std::true_type {};
+
+test constructible_policy_creatable_type = [] {
+  class must_be_creatable_or_bound : public di::config {
+   public:
+    static auto policies(...) noexcept {
+      using namespace di::policies;
+      using namespace di::policies::operators;
+
+      const auto is_bound_root_shared_ptr = is_root<di::_>{} && is_shared_ptr<di::_>{} && is_bound<di::_>{};
+      const auto is_bound_not_root = !is_root<di::_>{} && is_bound<di::_>{};
```

Comment:
This does work for my project but generates warning only. How can we make it error?

Reply:
It's always a warning, I think. You don't compile with `-Werror`?

Reply:
Unfortunately no :) may be one day...

Reply:
hmm, did you have errors before? I didn't change the reporting it should be a warning and a linker error (it's a trick to avoid long error messages showing all call stack)

Reply:
Yes, I am getting tons of errors but I was surprised this time it was one warning

Reply:
hmm, interesting, I'll check in msvc